.github/workflows/test.yml

```diff
@@ -33,3 +33,78 @@ jobs:
 
       - run: pip3 install hatch
       - run: CONFIG_ROOT_DIR=`pwd`"/dags" hatch run tests.py3.12-2.10:static-check
+
+  Run-Unit-Tests:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [ "3.8", "3.9", "3.10", "3.11", "3.12" ]
+        airflow-version: [ "2.2", "2.3", "2.4", "2.5", "2.6", "2.7", "2.8", "2.9", "2.10" ]
```

Comment:
If we are enabling older Airflow version do we need to allow older Python version too like Python3.6?

Reply:
I think we should not since Python 3.6 is already EOL. same for 3.7. 3.8 is EOL too, so wondering if we should remove that too :)

Reply:
The last release of DAG FActory (0.19.0) before the donation to Astronomer already only supported Python 2.7 and higher:
<img width="1624" alt="Screenshot 2024-10-15 at 10 08 21" src="https://github.com/user-attachments/assets/9ab6639f-12ae-4f60-851c-038c8633d590">

So it looks good - we're trying to keep it compatible with previous versions.

Reply:
It appears we decided to drop 3.7 in https://github.com/astronomer/dag-factory/pull/189 and continued doing so when migrated to use hatchling https://github.com/astronomer/dag-factory/pull/208. IMO, it would be wise if we do not add back 3.7. But please let me know if you feel otherwise.

Reply:
I am resolving this conversation as the PR needs all conversations to be resolved before merging, but please leave a comment on the PR if you've an opinion here, thanks.